packages/app-store/closecomothercalendar/lib/CalendarService.ts

```diff
@@ -0,0 +1,290 @@
+import { Credential } from "@prisma/client";
+
+import CloseCom, { CloseComCustomActivityCreate } from "@calcom/lib/CloseCom";
+import { symmetricDecrypt } from "@calcom/lib/crypto";
+import logger from "@calcom/lib/logger";
+import type {
+  Calendar,
+  CalendarEvent,
+  EventBusyDate,
+  IntegrationCalendar,
+  NewCalendarEventType,
+} from "@calcom/types/Calendar";
+
+const CALENDSO_ENCRYPTION_KEY = process.env.CALENDSO_ENCRYPTION_KEY || "";
+
+// Cal.com Custom Activity Fields
+const calComCustomActivityFields: [string, string, boolean, boolean][] = [
+  // Field name, field type, required?, multiple values?
+  ["Attendees", "contact", false, true],
+  ["Date & Time", "datetime", true, false],
+  ["Time zone", "text", true, false],
+  ["Organizer", "contact", true, false],
+  ["Additional notes", "text", false, false],
+];
+
+/**
+ * Authentication
+ * Close.com requires Basic Auth for any request to their APIs, which is far from
+ * ideal considering that such a strategy requires generating an API Key by the
+ * user and input it in our system. A Setup page was created when trying to install
+ * Close.com App in order to instruct how to create such resource and to obtain it.
+ *
+ * Meeting creation
+ * Close.com does not expose a "Meeting" API, it may be available in the future.
+ *
+ * Per Close.com documentation (https://developer.close.com/resources/custom-activities):
+ * "To work with Custom Activities, you will need to create a Custom Activity Type and
+ * likely add one or more Activity Custom Fields to that type. Once the Custom Activity
+ * Type is defined, you can create Custom Activity instances of that type as you would
+ * any other activity."
+ *
+ * Contact creation
+ * Every contact in Close.com need to belong to a Lead. When creating a contact in
+ * Close.com as part of this integration, a new generic Lead will be created in order
+ * to assign every contact created by this process, and it is named "From Cal.com"
+ */
```

Comment:
Maybe it would be a good idea to create a README file within the app package to explain all this

Reply:
Agreed